src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultConverters.cs

```diff
@@ -0,0 +1,303 @@
+                      options.GetAttributes<TAttribute>(parentClassType.Assembly).FirstOrDefault() ??
+                      options.GetAttributes<TAttribute>(JsonSerializerOptions.GlobalAttributesProvider).FirstOrDefault();
+            }
+
+            return attr;
+        }
+
+        public static TProperty GetPropertyValueOption<TProperty>(
+            Type parentClassType,
+            PropertyInfo propertyInfo,
+            JsonSerializerOptions options,
+            Func<JsonPropertyValueAttribute, TProperty> selector)
+        {
+            TProperty value;
+
+            JsonPropertyValueAttribute attr = null;
+            if (propertyInfo != null)
+            {
+                // Use Property first
+                attr = options.GetAttributes<JsonPropertyValueAttribute>(propertyInfo).FirstOrDefault();
+            }
+
+            if (attr == null || (value = selector(attr)) == default)
+            {
+                // Then class type
+                attr = options.GetAttributes<JsonPropertyValueAttribute>(parentClassType, inherit: true).FirstOrDefault();
+                if (attr == null || (value = selector(attr)) == default)
+                {
+                    // Then declaring assembly
+                    attr = options.GetAttributes<JsonPropertyValueAttribute>(parentClassType.Assembly).FirstOrDefault();
+
+                    if (attr == null || (value = selector(attr)) == default)
+                    {
+                        // Then global
+                        attr = options.GetAttributes<JsonPropertyValueAttribute>(JsonSerializerOptions.GlobalAttributesProvider).FirstOrDefault();
+
+                        if (attr == null)
+                        {
+                            value = default;
+                        }
+                        else
+                        {
+                            value = selector(attr);
+                        }
+                    }
+                }
+            }
+
+            return value;
+        }
+
+        public static JsonEnumerableConverter GetEnumerableConverter(
+            Type parentClassType,
+            PropertyInfo propertyInfo,
+            Type propertyType,
+            JsonSerializerOptions options)
+        {
+            Type enumerableType;
+            if (propertyType.IsGenericType)
+            {
+                enumerableType = propertyType.GetGenericTypeDefinition();
+            }
+            else if (propertyType.IsArray)
+            {
+                enumerableType = typeof(Array);
+            }
+            else
+            {
+                enumerableType = propertyType;
+            }
+
+            JsonEnumerableConverterAttribute attr = null;
+            if (propertyInfo != null)
+            {
+                // Use Property first
+                attr = options.GetAttributes<JsonEnumerableConverterAttribute>(propertyInfo).Where(a => a.EnumerableType == enumerableType).FirstOrDefault();
```

Comment:
I'm going to revisit this code once we agree on the use of attributes... However since the information is cached on the PropertyInfo<> it is a one-time hit for a given POCO type.